pkg/cmd/cli/cmd/deploy.go

```diff
@@ -221,10 +221,10 @@ func (o DeployOptions) deploy(config *deployapi.DeploymentConfig, out io.Writer)
 		}
 	}
 
-	config.LatestVersion++
+	config.Status.LatestVersion++
```

Comment:
This looks wrong.  Why would our client be updating a `Status` field?  It seems like we'd want to restructure this so that a client requests a redepoyment and a controller creates one.

@openshift/api-review This is a pre-existing problem and would be an API change that has consequences for backwards compatibility, but this decision here means that we can't trust this `Status` object to be accurate until we stop users from tweaking it.